### PR TITLE
Internal: Update wp env [ED-12013]

### PR DESCRIPTION
**Pull request by dmaziar**
## PR Checklist

- [ ] The commit message follows our guidelines:  https://github.com/elementor/elementor/blob/master/.github/CONTRIBUTING.md

## PR Type
What kind of change does this PR introduce?

- [ ] Bugfix
- [ ] Feature
- [ ] Code style update (formatting, local variables)
- [ ] Refactoring (no functional changes, no api changes)
- [ ] Build related changes
- [ ] CI related changes
- [ ] Documentation content changes
- [ ] Other... Please describe:

## Summary

This PR can be summarized in the following changelog entry:

*

## Description
An explanation of what is done in this PR

*

## Test instructions
This PR can be tested by following these steps:

*

## Quality assurance

- [ ] I have tested this code to the best of my abilities
- [ ] I have added unittests to verify the code works as intended
- [ ] Docs have been added / updated (for bug fixes / features)

Fixes #

### CODE REVIEW

File: .github/workflows/plugins-tester.yml [hunk 1]
```diff
@@ -106,12 +106,6 @@ jobs:
         run: |
             echo "PLUGINS_TESTER__PLUGINS_TO_TEST=${{ github.event.inputs.plugins }}" >> $GITHUB_ENV
             echo "PLUGINS_TESTER__DIFF_THRESHOLD=${{ github.event.inputs.diffThreshold }}" >> $GITHUB_ENV
-      - name: Run wp-env
-        uses: ./.github/workflows/run-wp-env
-        with:
-          PLUGINS: './tests/plugins-tester/plugins/elementor-tests-utils'
-          PORT: '7777' # In order to enable local run side by side with another default server (port 8888)
```
Comment (Barcov) [inline .github/workflows/plugins-tester.yml:113]: Is there a workaround to using this locally if we remove the port here?

Comment (dmaziar) [reply .github/workflows/plugins-tester.yml:113]: you can use it locally. It will take 7777 from wp-env.json. (In plugin tester folder). The problem is that you need a linux machine, otherwise screenshots will still have diffs.

File: tests/plugins-tester/src/PluginsTester.js [hunk 2]
```diff
@@ -35,27 +33,28 @@ export class PluginsTester {
 	}
 
 	runWP( cmd ) {
-		if ( ! this.options.runServer ) {
-			return this.cmd( `cd ../../ && ${ cmd }` );
-		}
+		// If ( ! this.options.runServer ) {
+		// 	return this.cmd( `cd ../../ && ${ cmd }` );
+		// }
```
Comment (yotamselementor) [inline tests/plugins-tester/src/PluginsTester.js]: Please remove commented-out code

File: tests/playwright/pages/wp-admin-page.ts [hunk 3]
```diff
@@ -190,8 +190,9 @@ export default class WpAdminPage extends BasePage {
 	}
 
 	async closeBlockEditorPopupIfVisible() {
```
Comment (TzviRabinovitch) [inline tests/playwright/pages/wp-admin-page.ts:192]: is this relevant to this PR ?

File: tests/playwright/sanity/core/base/document.test.ts [hunk 4]
```diff
@@ -50,7 +50,8 @@ test.describe( 'Document tests', async () => {
 } );
 
 async function addElement( wpAdmin: WpAdminPage, elementType: string ) {
-	await wpAdmin.page.click( '.block-editor-inserter__toggle' );
+	const frame = wpAdmin.page.frame( { name: 'editor-canvas' } );
```
Comment (TzviRabinovitch) [inline tests/playwright/sanity/core/base/document.test.ts:53]: same as above

File: .github/workflows/run-wp-env/action.yml [hunk 5]
```diff
@@ -39,16 +39,17 @@ runs:
     - name: Install wp-env
       shell: bash
       run: |
-          npm i -g --no-package-lock --no-save @wordpress/env@5.7.0
+          npm i -g --no-package-lock --no-save @wordpress/env@8.6.1
```
Comment (yotamselementor) [inline .github/workflows/run-wp-env/action.yml:42]: Lets have a single place to specify `wp-env` in (`package.json`)
```suggestion
          npm ci
```

File: .github/workflows/run-wp-env/action.yml [hunk 6]
```diff
@@ -39,16 +39,17 @@ runs:
...
     - name: Setup WordPress environment
       shell: bash
       run: |
         npx wp-env start
+        npx wp-env run cli chmod -R 777 wp-content
```
Comment (yotamselementor) [inline .github/workflows/run-wp-env/action.yml:47]: Why do we need this?

File: tests/plugins-tester/src/PluginsTester.js [hunk 7]
```diff
@@ -35,27 +33,28 @@ export class PluginsTester {
 	}
 
 	runWP( cmd ) {
-		if ( ! this.options.runServer ) {
-			return this.cmd( `cd ../../ && ${ cmd }` );
-		}
+		// If ( ! this.options.runServer ) {
```
Comment (TzviRabinovitch) [inline tests/plugins-tester/src/PluginsTester.js:36]: remove

**Review by TzviRabinovitch — DISMISSED** [review]

File: .github/workflows/lighthouse.yml [hunk 8]
```diff
@@ -89,13 +89,15 @@ jobs:
...
       - name: Run wp-env
         if: matrix.wpCoreVersion != 'master'
-        uses: ./.github/workflows/run-wp-env
-        with:
-          TEMPLATES: './tests/lighthouse/templates'
-          PHP_VERSION: ${{ matrix.phpVersion }}
-          WP_CORE_VERSION: ${{ matrix.wpCoreVersion }}
+        run: |
+          npx wp-env start
+          npx wp-env run cli bash elementor-config/setup.sh
```
Comment (yotamselementor) [inline .github/workflows/lighthouse.yml:100]: Lets use centralized scripts to avoid differences between CI and local development
```suggestion
          npm run start-local-server
          npm run test:setup:playwright
```

File: .github/workflows/lighthouse.yml [hunk 9]
```diff
@@ -89,13 +89,15 @@ jobs:
         with:
           path: ./build/*
           key: ${{ github.sha }}
+      - name: Install wp-env
+        shell: bash
+        run: |
+              npm i -g --no-package-lock --no-save @wordpress/env@8.6.1
```
Comment (yotamselementor) [inline .github/workflows/lighthouse.yml:95]: Lets have a single place to specify `wp-env` in (`package.json`)
```suggestion
          npm ci
```

File: .github/workflows/run-wp-env/action.yml [hunk 10]
```diff
@@ -39,16 +39,17 @@ runs:
...
         # change wp-env folder owner to www-data
         sudo chown 33:33 -R ~/wp-env/
```
Comment (yotamselementor) [inline .github/workflows/run-wp-env/action.yml:49]: I think this can be removed

File: .github/workflows/run-wp-env/action.yml
Comment (yotamselementor) [inline .github/workflows/run-wp-env/action.yml:1]: Do we really need this file anymore (as opposed to some simple script or two in package.json)?

**Review by yotamselementor — CHANGES_REQUESTED** [review]

File: tests/playwright/pages/wp-admin-page.ts (hunk 3 above)
Comment (dmaziar) [reply tests/playwright/pages/wp-admin-page.ts:192]: I think new wp-env is introducing a new changes. Add block icon is now in iframe, so I have to fix it

File: .github/workflows/run-wp-env/action.yml
Comment (dmaziar) [reply .github/workflows/run-wp-env/action.yml:1]: nope. I can remove it.

File: .github/workflows/run-wp-env/action.yml (hunk 5 above)
Comment (dmaziar) [reply .github/workflows/run-wp-env/action.yml:42]: I'll remove the file

File: .github/workflows/run-wp-env/action.yml (hunk 6 above)
Comment (dmaziar) [reply .github/workflows/run-wp-env/action.yml:47]: Wordress doesn't allow to write into this directory, so we can install plugin. Actually we have a lot of permissions problem if we use github action to create a new env. That is why I switch to npx wp-env start

File: tests/plugins-tester/src/PluginsTester.js (hunk 7 above)
Comment (dmaziar) [reply tests/plugins-tester/src/PluginsTester.js]: fixed

File: tests/plugins-tester/src/PluginsTester.js (hunk 2 above)
Comment (dmaziar) [reply tests/plugins-tester/src/PluginsTester.js]: done

File: .github/workflows/lighthouse.yml (hunk 8 above)
Comment (dmaziar) [reply .github/workflows/lighthouse.yml]: done

File: .github/workflows/lighthouse.yml (hunk 9 above)
Comment (dmaziar) [reply .github/workflows/lighthouse.yml]: done

File: .github/workflows/run-wp-env/action.yml (hunk 6 above)
Comment (yotamselementor) [reply .github/workflows/run-wp-env/action.yml]: Shouldn't we add it to the `start-local-server` script?

**Review by yotamselementor — APPROVED** [review]
